packages/react-flow-editor/src/Editor/components/Node/helpers.ts

```diff
@@ -7,3 +8,16 @@ export const nodeStyle = (pos: Point, state: NodeState | null) => ({
 })
 
 export const buildDotId = (nodeId: string) => `dot-${nodeId}`
+
+const isNoNodeFreeInputs = (node: Node, nodes: Node[]): boolean =>
+  nodes.filter((curNode) => curNode.outputs.map((out) => out.nextNodeId).includes(node.id)).length === node.inputNumber
+
+const isSameNode = (connectableNode: Node, curNode: Node): boolean => curNode.id === connectableNode.id
+
+export const markDisabledNodes = (connectableNode: Node, nodes: Node[]) => {
+  const disabledNodesIds = nodes
+    .filter((curNode) => !isSameNode(connectableNode, curNode) && isNoNodeFreeInputs(curNode, nodes))
```

Comment:
instead of negate condition you can change func `isSameNode`  to `notTheSameNode` using `!==`. It would be reading better